pkg/reconciler/serverlessservice/resources/services.go

```diff
@@ -98,15 +111,19 @@ func filterSubsetPorts(targetPort int32, subsets []corev1.EndpointSubset) []core
 	}
 	ret := make([]corev1.EndpointSubset, len(subsets))
 	for i, sss := range subsets {
-		sst := sss.DeepCopy()
+		sst := sss
+		sst.Ports = nil
 		// Find the port we care about and remove all others.
-		for j, p := range sst.Ports {
+		for j, p := range sss.Ports {
 			if p.Port == targetPort {
-				sst.Ports = sst.Ports[j : j+1]
-				break
+				sst.Ports = append(sst.Ports, sss.Ports[j])
+			}
+			// Add HTTPS port if exists.
+			if p.Port == networking.BackendHTTPSPort {
+				sst.Ports = append(sst.Ports, sss.Ports[j])
 			}
```

Comment:
One more minor improvement if you wish:

```go
switch p.Port {
case targetPort, networking.BackendHTTPSPort:
  sst.Ports = append(sst.Ports, sss.Ports[j])
}
```

You could also have two `case` statements if you wish, as:

```go
switch p.Port {
case networking.BackendHTTPSPort:
  fallthrough
case targetPort:
  sst.Ports = append(sst.Ports, sss.Ports[j])
}
```